### PR TITLE
DBZ-7345 Oracle log de-dupe on sequence and redo thread

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-7345

@jpechane, this fix supplements the already existing fixes we made for this on Debezium 2.3, 2.4, and 2.5 to handle a corner case where on Oracle RAC another node could be using the same log sequence and the deduplication phase of archive and redo logs could mistakenly remove more logs than necessary.

### CODE REVIEW

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java [hunk 1]
```diff
@@ -205,21 +206,28 @@ else if ("ONLINE".equals(type)) {
...
         // To avoid duplicate log files (ORA-01289 cannot add duplicate logfile)
-        // Remove the archive log which has the same sequence number.
-        for (LogFile redoLog : onlineLogFiles) {
-            archivedLogFiles.removeIf(f -> {
-                if (f.getSequence().equals(redoLog.getSequence())) {
-                    LOGGER.debug("Removing archive log {} with duplicate sequence {} to {}", f.getFileName(), f.getSequence(), redoLog.getFileName());
+        // Remove the archive log which has the same sequence number and redo thread number.
+        for (LogFile redoLog : redoLogs) {
+            archiveLogs.removeIf(f -> {
+                if (f.getSequence().equals(redoLog.getSequence()) && f.getThread() == redoLog.getThread()) {
```
Comment (jpechane) [inline debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java:221]: @Naros Does it make sense to introduce a comparison method directly on `LogFile` class?

**Comment by obabec** [comment]
@Naros oracle 21 is still failing :(

**Comment by Naros** [comment]
> @Naros oracle 21 is still failing :(

Ya, I know why too -- just need to look at the test and fix it.

File: debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java (hunk 1 above)
Comment (Naros) [reply debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerHelper.java]: Used the existing `#equals` method that had the same logic.

**Comment by jpechane** [comment]
@Naros Applied, thanks!

**Pull request merged**